GameAI/behaviour/Complex/ObstacleAvoidance.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GameAI.entity;
+using Microsoft.Xna.Framework;
+using MonoGame.Extended;
+
+namespace GameAI.behaviour.Complex
+{
+    class ObstacleAvoidance : SteeringBehaviour
+    {
+        private int range;
+        private List<BaseGameEntity> obstacles;
+
+        public ObstacleAvoidance(MovingEntity entity, List<BaseGameEntity> obstacles, int range = 100) : base(entity)
```

Comment:
Waarom een List ipv een IEnumerable? We hoeven niets toe te voegen, toch? 

Overigens is het voor Spatial Partitioning beter om om een instantie van de World-klasse te vragen.